vendor/github.com/waku-org/go-waku/waku/v2/api/history/cycle.go

```
@@ -98,6 +98,12 @@ func (m *StorenodeCycle) DisconnectActiveStorenode(backoff time.Duration) {
 func (m *StorenodeCycle) connectToNewStorenodeAndWait(ctx context.Context) error {
 	// Handle pinned storenodes
 	m.logger.Info("disconnecting storenode")
+
+	if m.storenodeConfigProvider == nil {
+		m.logger.Debug("storenodeConfigProvider not yet setup")
+		return nil
```

Shall return an error instead?
Maybe not needed but I think is better to make the error message different so that it is easier to determine when it happened

---

Yes! But notice that these error handling improvements are in the `go-waku` vendor, for already merged code - I only updated the vendor here.

I think we should rather focus on `status-go` and `nwaku` code and not focus on improving `go-waku` as it has its days counted

---

Ooh yes, you are right. Makes 100% sense, thanks